unifi_console/datadog_checks/unifi_console/check.py

```diff
@@ -0,0 +1,111 @@
+from typing import cast
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.unifi_console.api import APIConnectionError, UnifiAPI
+from datadog_checks.unifi_console.config import UnifiConfig
+from datadog_checks.unifi_console.mertrics import Metric
+from datadog_checks.unifi_console.types import Check, ControllerInfo
+
+
+class UnifiConsoleCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = "unifi_console"
+
+    def __init__(self, name, init_config, instances):
+        instances[0]['persist_connections'] = instances[0].get('persist_connections', True)
+        super(UnifiConsoleCheck, self).__init__(name, init_config, instances)
+
+        # Use self.instance to read the check configuration
+        self._config = UnifiConfig(self.instance, self.init_config, self.log)
+        self.api = UnifiAPI(self._config, self.http, self.log)
+
+        # try to login a initialization to prevent login for every request
+        self.check_initializations.append(self._initiate_api_connection)
+
+    def _initiate_api_connection(self):
+        try:
+            self.log.debug(
+                "Connecting to the Unifi Controller Api %s with username %s...",
+                self._config.url,
+                self._config.user,
+            )
+            self.api.connect()
+            self.log.debug("Connected")
+        except APIConnectionError:
+            self.log.error("Cannot authenticate to Unifi Controller API. The check will not run.")
+            self.service_check(
+                "can_connect",
+                AgentCheck.CRITICAL,
+                tags=self._config.tags,
+                hostname=None,
+            )
+            raise
+
+    def check(self, _):
+
+        # Assert the health of the Unifi API by getting the status, and submit the service_check accordingly
+        status = cast(ControllerInfo, None)
+        try:
+            status = self.api.status()
+            print(status)
```

Comment:
```suggestion
            print(status)
```
Leftover from troubleshooting?